Modules/cjkcodecs/multibytecodec.c

```diff
@@ -895,6 +901,76 @@ _multibytecodec_MultibyteIncrementalEncoder_encode_impl(MultibyteIncrementalEnco
     return encoder_encode_stateful(STATEFUL_ECTX(self), input, final);
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_getstate_impl(MultibyteIncrementalEncoderObject *self)
+/*[clinic end generated code: output=9794a5ace70d7048 input=4a2a82874ffa40bb]*/
+{
+    const char *pendingbuffer = NULL;
+    Py_ssize_t pendingsize;
+    MultibyteIncrementalEncoderState state;
+
+    memset(&state, 0, sizeof(state));
+    if (self->pending != NULL) {
+        pendingbuffer = PyUnicode_AsUTF8AndSize(self->pending, &pendingsize);
+        if (pendingbuffer == NULL) {
+            return NULL;
+        }
+        memcpy(state.pending, pendingbuffer, pendingsize);
+        state.pendingsize = pendingsize;
+    }
+    state.codecstate = self->state;
+
+    return _PyLong_FromByteArray((const unsigned char *)&state,
+                                 sizeof(state),
+                                 PY_LITTLE_ENDIAN, 0);
+}
+
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.setstate
+    statelong: object(type='PyLongObject *', subclass_of='&PyLong_Type')
+    /
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_setstate_impl(MultibyteIncrementalEncoderObject *self,
+                                                          PyLongObject *statelong)
+/*[clinic end generated code: output=4e5e98ac1f4039ca input=9c9c5abfe365c1c1]*/
+{
+    PyObject *pending = NULL;
+    MultibyteIncrementalEncoderState state;
+
+    if (_PyLong_AsByteArray(statelong, (unsigned char *)&state,
+                            sizeof(state),
+                            PY_LITTLE_ENDIAN, 0) < 0) {
+        goto errorexit;
+    }
+
+    if (state.pendingsize > MAXENCPENDING) {
```

Comment:
MAXPENDING*4?

Reply:
Ah yes. Updated the test to check the new boundary too.